### PR TITLE
[POC] next-css

### PR DESCRIPTION
**Pull request by mwickett**
Took [`zeit/next-css`](https://github.com/zeit/next-plugins/tree/master/packages/next-css) for a spin. It's pretty basic, but it works.

We don't get the generated scoped class names (we'd need to transition to using CSS Modules for that to work with this package – too big of a change IMO), but we don't have that now, and using GPS has given us the right level of scoping. I'm not totally sure that we need the rando class names? 

It's a little bit magical just dropping `import css from './styles.css` at the top of a component, but not too bad. Global styles import fine at a top level component. PostCSS config works as expected.

Overall this feels much closer to what we have with our current setup now - more on the CSS end of the CSS <---> JS spectrum.

For discussion!

### CODE REVIEW

File: components/global-styles/variables.global.css [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  --nomad-green-dark: #005738;
+  --terraform-purple: #5f43e9;
+  --teraform-purple-dark: #2a1c73;
+  --consul-pink: #ca2171;
+  --consul-pink-dark: #650d34;
+  --vault-gray: #797e8d;
+  --vault-gray-dark: #22242e;
+  --packer-blue: #00acff;
+  --packer-blue-dark: #005283;
+}
```
Comment (jescalan) [inline components/global-styles/variables.global.css]: Curious about this, why was it switched over? Can we import once and have it be available for all stylesheets, or do we need to import every time it's used? How does it output?

Comment (mwickett) [reply components/global-styles/variables.global.css]: Good question. It was working, but for some reason importing variables (as a CSS file) into `_app.js` doesn't work, unless `preserve: true` is on for PostCSS custom properties.

I've switched it back to using the `variables.js` approach with `importFrom` and things are working well.

**Comment by mwickett** [comment]
This could be a major problem: https://github.com/zeit/next-plugins/issues/1

**Comment by mwickett** [comment]
I am going to test this with a more complex component that has nested components.

**Comment by mwickett** [comment]
I've updated this to include case-study-slider and things seem to be working nicely - global styles are imported at `_app.js` and component styles just need to be imported once, no `<style jsx>` tags needed. Also simplifies our webpack config.

Thoughts?

**Comment by jescalan** [comment]
Ok I think we should merge this and move forward here we go!

**Pull request merged**